job-runner/chain/traverser.go

```diff
@@ -431,6 +431,10 @@ func (t *traverser) runJobs() {
 			// Increment sequence try count if this is sequence start job, which
 			// currently means sequenceId == job.Id.
 			if t.chain.IsSequenceStartJob(job.Id) {
+				if t.chain.SequenceTries(job.Id) != 0 {
+					retryWait, _ := time.ParseDuration(job.SequenceRetryWait) // checked that this parses in RM
+					time.Sleep(retryWait)
```

Comment:
Ah, if the traverser gets stopped or shutdown (Stop() / shutdown()) while we're doing a sequence retry wait, it'll need to wait for this sleep to finish before it can finish actually stopping the traverser (and it times out after 10 seconds or something so that's not great). Unfortunately the whole stopping / suspending thing is really complicated. Anyways, I think the fix for this is:
1. move this `if` block to _before_ the defer at line 418.
2. replace time.Sleep with something like
```
select {
    case <-time.After(retryWait):
        # do nothing, we waited successfully
    case <-t.stopChan:
        jLogger.Infof("traverser was stopped - exiting sequence retry wait early and not running job")
        // before returning, decrement pending to signal to stopRunningJobs that there's
        // one less goroutine it nees to wait for.
        atomic.AddInt64(&t.pending, -1)
        return
}
```